src/udev/udev-manager.c

```diff
@@ -908,7 +918,37 @@ static int synthesize_change(sd_device *dev) {
         if (startswith(sysname, "dm-") || block_device_is_whole_disk(dev) <= 0)
                 return synthesize_change_one(dev, dev);
 
-        return synthesize_change_all(dev);
+        _cleanup_(pidref_done) PidRef pidref = PIDREF_NULL;
+        r = pidref_safe_fork(
+                        "(udev-synth)",
+                        FORK_RESET_SIGNALS|FORK_CLOSE_ALL_FDS|FORK_DEATHSIG_SIGTERM|FORK_LOG|FORK_RLIMIT_NOFILE_SAFE,
+                        &pidref);
+        if (r < 0)
+                return r;
+        if (r == 0) {
+                /* child */
+                (void) synthesize_change_all(dev);
+                _exit(EXIT_SUCCESS);
+        }
+
+        _cleanup_(sd_event_source_unrefp) sd_event_source *s = NULL;
+        r = event_add_child_pidref(manager->event, &s, &pidref, WEXITED, synthesize_change_child_handler, manager);
+        if (r < 0) {
+                log_debug_errno(r, "Failed to add child event source for "PID_FMT", ignoring: %m", pidref.pid);
+                return 0;
+        }
+
+        r = sd_event_source_set_child_pidfd_own(s, true);
+        if (r < 0)
+                return r;
+        TAKE_PIDREF(pidref);
```

Comment:
one of those days we should add a wrapper for pidref_safe_fork() that returns an sd_event_source object in one go.